src/docs/asciidoc/Modules.asciidoc

```diff
@@ -105,40 +121,44 @@ buildscript {
 
-* The `MODULE` layout for Spring Boot packaging ensures `provided` dependencies will not be included in the uber-jar. The Spring XD module build support declares `spring-xd-dirt` as a provided dependency, as some of its classes are needed for module development.
-* Any compile dependencies, transitive or declared for the module will be excluded from the uber-jar if they are also Spring XD runtime dependencies.
+* The `MODULE` layout for Spring Boot packaging ensures `provided` dependencies will not be included in the uber jar. The Spring XD module tools declare `spring-xd-dirt` as a provided dependency, as some of its classes are needed for module development.
+* Any compile or runtime dependencies, transitive or explicitly declared, for the module will be excluded from the uber jar if they are also Spring XD runtime dependencies.
 
-[NOTE]
-====
-In rare cases, it may be necessary to override the default exclusions. For example, if your module requires a different version of library that is on the Spring XD class path, you can override the boot maven plugin configuration in your pom, like so:
+
+The build support tools declare dependencies on `spring-xd-dirt` and `spring-xd-test` which provide some useful features for module development, including support for:
+
+* Java defined xref:ModuleOptionsMetadata#module-options-metadata[Module Options Metadata]
+* In-container module testing - start an embedded single node container, deploy your module and validate the results.
+
+[[module-dependency-management]]
+==== Module dependency management
+
+Runtime issues may arise if module dependencies conflict with those on the Spring XD classpath. This happens when a module declares a dependency using a different version of a jar provided by the Spring XD runtime. If the versions are the same, the build tools will ensure that these dependencies will not be exported to the module jar. However if the versions declared in your build are different, the dependencies will be exported and loaded by the module class loader. The tools are designed to permit you to override the version, but generally we don't recommend this unless you have a specific requirement to do so. If there are conflicting dependencies, your module may or may not work. Symptoms of version conflicts include `ClassDefNotFoundException` or `NoSuchMethodError` when you deploy the module.  If you encounter such errors, you should manually check the contents of `$XD_INSTALL_DIR/xd/lib` against the contents of the module jar, and make the necessary changes to your build script. Compile time dependencies must be declared in any case. If they are also on the Spring XD classpath they should be declared as `provided`. 
```

Comment:
I should clarify this. It's true for Gradle, not Maven which is explained further down.